test/ParseableInterface/convenience-init.swift

```diff
@@ -0,0 +1,51 @@
+// RUN: %empty-directory(%t)
+
+// Generate the parseable interface of the current file via the merge-modules step
+// RUN: %target-build-swift -emit-module -o %t/Test.swiftmodule -emit-parseable-module-interface-path %t/TestMerge.swiftinterface -module-name Test %s
+
+// Generate the parseable interface of the current file via a single frontend invocation
+// RUN: %target-swift-frontend -typecheck  -enable-objc-interop -emit-parseable-module-interface-path %t/TestSingle.swiftinterface -module-name Test %s
+
+// Make sure both don't add override for inits shadowing convenience initializers
+// RUN: %FileCheck --check-prefixes=CHECK,SINGLE %s < %t/TestSingle.swiftinterface
+// RUN: %FileCheck --check-prefixes=CHECK,MERGE %s < %t/TestMerge.swiftinterface
+
+// Check we can consume the interface without issue
+// RUN: %target-swift-frontend -swift-version 5 -build-module-from-parseable-interface -o %t/Test.swiftmodule %t/TestSingle.swiftinterface
+// RUN: %target-swift-frontend -swift-version 5 -build-module-from-parseable-interface -o %t/Test.swiftmodule %t/TestMerge.swiftinterface
+
+public class Base {
+  let x: Int
+  public init(x: Int) {
+    self.x = x
+  }
+  convenience public init() {
+    self.init(x: 1)
+  }
+}
+
+public class Derived: Base {
+  // MERGE: {{^}}  public init(z: Swift.Int)
+  // SINGLE: {{^}}  public init(z: Int)
+  public init(z: Int) {
+    super.init(x: z)
+  }
+  // MERGE: {{^}}  public convenience init()
+  // SINGLE: {{^}}  convenience public init()
```

Comment:
This switched order here is because we have a 'ConvenienceAttr` on the initializer when generating via the single invocation (which is printed when printing the decl's attributes) and don't after serializing/deserializing (where we print it based on the initializer kind).